pages/docs/couple-your-code/couple-your-code-waveform.md

```diff
@@ -0,0 +1,167 @@
+precice.finalize();
+delete[] vertexIDs, forces, displacements;
+turnOffSolver();
+```
+
+### Use waveform iteration with subcycling
+
+Another example shows how to use subcycling with the waveform iteration API. This is an extension of the example from ["Step 5 - Non-matching time step sizes"](couple-your-code-timestep-sizes.html). The changes are similar to the example above, but additionally the behavior of `isReadDataAvailable` slightly changes, because every time step now provides updated read data:
+
+```cpp
+while (not simulationDone()){ // time loop
+  dt = beginTimeStep(); // e.g. compute adaptive dt 
+  dt = min(precice_dt, dt);
+  if (precice.isReadDataAvailable()){ // always returns true with new API
+    // sampling in the middle of the time step
+    precice.readBlockVectorData(displID, vertexSize, vertexIDs, 0.5 * dt, displacements);
+    setDisplacements(displacements); // displacement at the middle of the time step
+  }
+  solveTimeStep(dt); // might be using midpoint rule for time-stepping
+  if (precice.isWriteDataRequired(dt)){ // only true at the end of the time window
+    computeForces(forces);
+    precice.writeBlockVectorData(forceID, vertexSize, vertexIDs, forces);
+  }
+  precice_dt = precice.advance(dt);
+  endTimeStep(); // e.g. update variables, increment time
+}
+```
+
+### Note on the use of `initializeData`
+
+{% include important.html content="`writeBlockVectorData` *after* `initializeData` writes data corresponding to the *end* of the window." %}
+
+In the very first time window of our coupled simulation we have to provide initial data in order to be able to perform linear interpolation. We have to call `writeBlockVectorData` *before* `initializeData`. This data will be used as initial data for the interpolation. If no initial data is provided, only constant interpolation can be applied in the first window, but linear interpolation is still available in later time windows.
+
+```cpp
+...
+// sets forces at beginning of the first time window, similar for displacement on the other participant's side.
+precice.writeBlockVectorData(forceID, vertexSize, vertexIDs, forces);
+precice.initializeData();
+precice_dt = precice.initialize();
+
+while (not simulationDone()){ // time loop
+  if (precice.isReadDataAvailable()){ // always returns true with new API
+    // sampling from interpolant in the middle of the time step
+    precice.readBlockVectorData(displID, vertexSize, vertexIDs, 0.5 * dt, displacements);
+  }
+  ...
+  if (precice.isWriteDataRequired(dt)){ // only true at the end of the time window
+    // sets forces at the end of the current time window
+    precice.writeBlockVectorData(forceID, vertexSize, vertexIDs, forces);
+  }
+  precice_dt = precice.advance(dt);
+  ...
+}
+```
+
+{% include note.html content="Depending on the coupling scheme, the used interpolation scheme might differ depending on the order of the participants and depending on whether we are in the first or a later coupling iteration." %}
```

Comment:
By "order" here you mean whether a participant is first or second, right? Might be confusing.
Overall, I find this statement confusing. We should be able to explain these cases. If not, we are doing sth wrong.

Reply:
I think this statement is not necessary anymore, because we agreed to exclude explicit coupling. For implicit coupling the execution order of participants does not matter (besides the first iteration, but I don't think we have to mention this).

Reply:
Revised this in 8ba2573eab4a20317a4cc5c247d21438d9f8f284.